File: dotnet/samples/01-get-started/01_HelloAgent.cs
// Copyright (c) Microsoft. All rights reserved.

// Step 1: Your First Agent
// The simplest possible agent — send a message, get a response.
// Uses Azure AI Foundry Responses API as the default provider.
//
// For more on agents, see: ../02-agents/README.md
// For docs: https://learn.microsoft.com/agent-framework/get-started/your-first-agent

using Azure.AI.Projects;
using Azure.AI.Projects.OpenAI;
using Azure.Identity;
using Microsoft.Agents.AI;

// <create_agent>
string endpoint = Environment.GetEnvironmentVariable("AZURE_FOUNDRY_PROJECT_ENDPOINT")
    ?? throw new InvalidOperationException("AZURE_FOUNDRY_PROJECT_ENDPOINT is not set.");
string deploymentName = Environment.GetEnvironmentVariable("AZURE_FOUNDRY_PROJECT_DEPLOYMENT_NAME") ?? "gpt-4o-mini";

AIProjectClient aiProjectClient = new(new Uri(endpoint), new AzureCliCredential());
AIAgent agent = await aiProjectClient.CreateAIAgentAsync(
    name: "Assistant",
    model: deploymentName,
    instructions: "You are a helpful assistant.");
// </create_agent>

// <run_agent>
Console.WriteLine(await agent.RunAsync("What is the capital of France?"));
// </run_agent>

// <run_agent_streaming>
await foreach (AgentResponseUpdate update in agent.RunStreamingAsync("Tell me a fun fact about Paris."))
{
    Console.Write(update);
}
Console.WriteLine();
// </run_agent_streaming>

// Cleanup
await aiProjectClient.Agents.DeleteAgentAsync(agent.Name);

File: python/samples/01-get-started/01_hello_agent.py
# Copyright (c) Microsoft. All rights reserved.

import asyncio
import os

from agent_framework.openai import OpenAIResponsesClient

"""
Hello Agent — Simplest possible agent

This sample creates a minimal agent using OpenAIResponsesClient and runs it
in both non-streaming and streaming modes.
"""


async def main() -> None:
    # <create_agent>
    client = OpenAIResponsesClient(
        api_key=os.environ["OPENAI_API_KEY"],
        model_id=os.environ.get("OPENAI_RESPONSES_MODEL_ID", "gpt-4o"),
    )

    agent = client.as_agent(
        name="HelloAgent",
        instructions="You are a friendly assistant. Keep your answers brief.",
    )
    # </create_agent>

    # <run_agent>
    # Non-streaming: get the complete response at once
    result = await agent.run("What is the capital of France?")
    print(f"Agent: {result}")
    # </run_agent>

    # <run_agent_streaming>
    # Streaming: receive tokens as they are generated
    print("Agent (streaming): ", end="", flush=True)
    async for chunk in agent.run("Tell me a one-sentence fun fact.", stream=True):
        if chunk.text:
            print(chunk.text, end="", flush=True)
    print()
    # </run_agent_streaming>


if __name__ == "__main__":
    asyncio.run(main())
